File: algo-competitions/starter128-div3/Moody_Chef.py
t = int(input())


def score(n: int, l, r):
    if l <= n <= r:
        return 1
    return -1


for _ in range(t):
    n, l, r = map(int, input().split())
    a = input().split()
    current, min, max = 0, 0, 0
    for i in a:
        current += score(int(i), l, r)
        if current > max:
            max = current
        if current < min:
            min = current
    print(max, min)

File: algo-competitions/starter128-div3/Moody_Chef.cpp
#include <iostream>
using namespace std;

int main()
{
    int t, n, l, r;
    cin >> t;
    for (; t > 0; t--)
    {
        int min = 0, max = 0, happiness = 0, tmp;
        cin >> n >> l >> r;
        for (int i = 0; i < n; i++)
        {
            cin >> tmp;
            if (l <= tmp and tmp <= r)
                happiness += 1;
            else
                happiness -= 1;

            if (happiness > max)
                max = happiness;
            if (happiness < min)
                min = happiness;
        }
        cout << max << " " << min << "\n";
    }
}